### DISCUSSION TITLE
keyCopyAllMeta

### DISCUSSION THREAD

**Discussion by markus2330** [ElektraInitiative/libelektra#3684] [Q&A]
> The only remaining questions from the review would be:

I nearly didn't see these questions. It is better if you start discussions in separate threads (as I did now).

> Is keyCopyAllMeta() still needed or should we deprecate it in favour of keyCopy(..., KEY_CP_META)? Are they even exactly interchangeable?

Unfortunately, they do not have the same behavior, which is a problem as with the same name (copy) you would expect them to behave the same.

I think we should reduce key*Meta functions as much as possible, keyCopyAllMeta is the first candidate to be removed. It is used nearly nowhere and can be easily implemented yourself using `keyMeta`.

> Should dest be added to the glossary? Although this is not specific to this function, and could easily be resolved in a future PR...

It does not harm, https://manned.org/strcpy.3 says several time that dest is a "destination string".

**Comment by kodebach** [comment]
> I nearly didn't see these questions. It is better if you start discussions in separate threads (as I did now).

Similarly, you should tag the appropriate people, or post a reference in the old thread, when moving to a new thread.

I assume this is addressed at me and @lawli3t ?

I already posted a response (https://github.com/ElektraInitiative/libelektra/pull/3606#issuecomment-788283051), but I agree removing `keyCopyAllMeta` in favour of `ksAppend(keyMeta(dest), keyMeta(source))` would be a good idea.
